hudi-common/src/main/java/org/apache/hudi/common/util/ReflectionUtils.java

```diff
@@ -90,16 +99,58 @@ public static Object loadClass(String clazz, Object... constructorArgs) {
   }
 
   /**
-   * Return stream of top level class names in the same class path as passed-in class.
-   * 
-   * @param clazz
+   * Scans all classes accessible from the context class loader
+   * which belong to the given package and subpackages.
+   *
+   * @param clazz class
+   * @return Stream of Class names in package
    */
-  public static Stream<String> getTopLevelClassesInClasspath(Class clazz) {
+  public static Stream<String> getTopLevelClassesInClasspath(Class<?> clazz) {
```

Comment:
this method is just used by the dummy bundle main classes right.. Ideally, like to simplify this down the line. if you agree, we can raise a JIRA?

Reply:
yes - let's raise a jira